stories/Autosuggest.js

```diff
@@ -52,85 +65,79 @@ storiesOf('Organisms|Autosuggest', module)
+        };
 
-            const onClearAllSelected = () => {
-                console.log('onClearAllSelected was called');
-                setSelectedSuggestions([]);
-            };
+        const onSelectionChange = item => {
+            console.log(`onSelectionChange was called with {name: ${item.name}}`);
+            if (store.get('selectedSuggestions').includes(item)) {
+                const newSelection = store
+                    .get('selectedSuggestions')
+                    .filter(el => el.name !== item.name);
+                store.set({ selectedSuggestions: newSelection });
+            } else {
+                store.set({ selectedSuggestions: [...store.get('selectedSuggestions'), item] });
+            }
+        };
 
-            return (
-                <Autosuggest
-                    selectedSuggestions={
-                        boolean('Add selectedSuggestions', true) ? selectedSuggestions : undefined
-                    }
-                    selectedPlaceholder={
-                        boolean('Add selectionPlaceholder', true)
-                            ? getSelectedPlaceholder()
-                            : undefined
-                    }
-                    isLoading={boolean('Loading', false)}
-                    showClearButton={boolean('Show clear button', true)}
-                    isMultiselect={boolean('Multiselect', true)}
-                    isProminent={boolean('Use prominent styling', true)}
-                    inputPlaceholder={text('Input placeholder', 'Select something...')}
-                    noSuggestionsPlaceholder={text('No suggestions', 'No suggestions found...')}
-                    clearTitle={text('Remove button label', 'Clear')}
-                    getSuggestions={getSuggestions}
-                    suggestionToString={SUGGESTION_TO_STRING}
-                    onBlur={onBlur}
-                    onSelectionChange={onSelectionChange}
-                    onClearAllSelected={onClearAllSelected}
-                    onInputValueChange={onInputValueChange}
-                    style={style}
-                />
-            );
+        const onBlur = () => {
+            console.log('onBlur was called');
+            store.set({ inputValue: '' });
         };
 
-        Implementation.displayName = 'Implementation';
+        const onClearAllSelected = () => {
+            console.log('onClearAllSelected was called');
+            store.set({ selectedSuggestions: [] });
+        };
 
-        return <Implementation style={{ width: '650px' }} />;
+        return (
+            <Autosuggest
+                selectedSuggestions={
+                    boolean('Add selectedSuggestions', true)
+                        ? store.get('selectedSuggestions')
+                        : undefined
+                }
+                selectedPlaceholder={
+                    boolean('Add selectionPlaceholder', true) ? getSelectedPlaceholder() : undefined
+                }
```

Comment:
Maybe we can add an icon with this same trick to render or not depending on the knob.

Reply:
are you mean something like this?
`
iconNode={
                        boolean('Add iconNode', true)
                            ? <IconMatch />
                            : undefined
                    }
`

Reply:
yes. Then all props are covered